tools/tiara/tiara.xml

```diff
@@ -78,38 +77,47 @@
         </section>
     </inputs>
     <outputs>
-        <collection name="output" type="list" label="${tool.name} on ${on_string}: classified sequences in txt and Fasta Output">
-            <discover_datasets pattern="__name_and_ext__"  ext="fasta,txt" directory="results" />
+        <collection name="output_fasta" type="list" label="${tool.name} on ${on_string}: Fasta Output">
+            <discover_datasets pattern="__name_and_ext__"  ext="fasta" directory="results" />
+        </collection>
+        <collection name="output_txt" type="list" label="${tool.name} on ${on_string}: Classified sequences in txt">
+            <discover_datasets pattern="__name_and_ext__"  ext="txt" directory="results" />
         </collection>
     </outputs>
     <tests>
-        <test expect_num_outputs="1">
+        <test expect_num_outputs="2">
             <param name="input" value="plast_fr.fasta.gz"/>
             <param name="taxonomy_filter" value="pla"/>
-            <output_collection name="output" type="list">
-                <element name="main_result" file="main_result01.txt" ftype="txt"/>
+            <output_collection name="output_fasta" type="list">
```

Comment:
you can count here the number of expected elements.

I think it will fail then, because you still collect all the elements.